### PR TITLE
tracing: Add thrift integration

### PR DESCRIPTION
**Pull request by fishy**
Add helper functions to create span from thrift context and to create
context from span.

Please note that this is a breaking change as we move away from
opentracing abstraction to use zipkin-go directly.

### CODE REVIEW

**Comment by fishy** [comment]
:haircut: @lbelaus 

As discussed on slack, I made changes to move away from opentracing abstraction to use zipkin library directly. Also added tests and moved string set to its own package. PTAL.

**Comment by fishy** [comment]
👓 @andizzle

File: set/string_test.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+func TestStringSet(t *testing.T) {
+	const (
+		key1 = "key1"
+		key2 = "key2"
+	)
+
+	var s set.String
+
+	t.Run(
```
Comment (konradreiche) [inline set/string_test.go]: Is that an alternative to table-driven testing to make debugging of tests easier?

Comment (fishy) [reply set/string_test.go]: This is more like control the order of sequential (sub-)tests, and also share state between them.

**Pull request merged**